dotCMS/src/main/java/com/dotcms/rest/IntegrityResource.java

```diff
@@ -153,68 +149,99 @@ private Response postWithEndpointState(String endpointId, String url, MediaType
 		return response;
 	}
 
+    /**
+     * Resolves remote IP address from request.
+     * @param request {@link HttpServletRequest}
+     * @return a String representing the remote IP address (or hostname)
+     */
+    private static String resolveRemoteIp(@Context final HttpServletRequest request) {
+        final String remoteIP = request.getRemoteHost();
+        return !UtilMethods.isSet(remoteIP) ? remoteIP : request.getRemoteAddr();
+    }
+
+    private static String getFullLocalIp(@Context final HttpServletRequest request) {
+        final String localIp = request.getLocalName();
+        Optional<String> port = HttpRequestDataUtil.getServerPort();
+        return (!UtilMethods.isSet(localIp) ? localIp : request.getLocalName())
+                + ':' + port.orElse(request.getLocalPort() + "");
+    }
 
     /**
      * <p>Returns a zip with data from structures and folders for integrity check
-     *
-     * Usage: /getdata
-     *
      */
-
     @POST
     @Path("/generateintegritydata/{params:.*}")
     @Consumes(MediaType.MULTIPART_FORM_DATA)
     @Produces("text/plain")
-    public Response generateIntegrityData(@Context HttpServletRequest request, @FormDataParam("AUTH_TOKEN") String auth_token_digest)  {
+    public Response generateIntegrityData(
+            @Context final HttpServletRequest request,
+            @FormDataParam("AUTH_TOKEN") final String auth_token_digest)  {
```

Comment:
Codacy found an issue: [The String literal "AUTH_TOKEN" appears 8 times in this file; the first occurrence is on line 169](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6151243)